airflow/task/task_runner/base_task_runner.py

```diff
@@ -133,16 +138,28 @@ def run_command(self, run_with=None):
 
         self.log.info("Running on host: %s", get_hostname())
         self.log.info('Running: %s', full_cmd)
-        # pylint: disable=subprocess-popen-preexec-fn,consider-using-with
-        proc = subprocess.Popen(
-            full_cmd,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-            universal_newlines=True,
-            close_fds=True,
-            env=os.environ.copy(),
-            preexec_fn=os.setsid,
-        )
+
+        if is_windows():
+            # pylint: disable=subprocess-popen-preexec-fn,consider-using-with
+            proc = subprocess.Popen(
+                full_cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                universal_newlines=True,
+                close_fds=True,
+                env=os.environ.copy()
+            )
+        else:
+            # pylint: disable=subprocess-popen-preexec-fn,consider-using-with
+            proc = subprocess.Popen(
+                full_cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                universal_newlines=True,
+                close_fds=True,
+                env=os.environ.copy(),
+                preexec_fn=os.setsid,       # does not exist on Windows
+            )
```

Comment:
I think we can use `start_new_session=True` instead. This options is a shorthand to `preexec_fn=os.setsid` if available, and silently ignored on Windows (if I’m understanding the documentation

Reply:
Seems to work, I will change it. Thanks!